metric_map.go

```diff
@@ -46,7 +50,10 @@ func (mm *MetricMap) Receive(m *Metric) {
 }
 
 func MergeMaps(mms []*MetricMap) *MetricMap {
-	mm := NewMetricMap()
+	if len(mms) == 0 { // garbage in, garbage out
+		return nil
+	}
+	mm := NewMetricMap(mms[0].Forwarded)
```

Comment:
Would we only want the first element here?

Reply:
Not 100% sure what you mean.

We assume that all maps have the same forwarded value.  If they don't then it's a logic bug somewhere else.  I don't want to deal with returning an error from this, so we just assume the first thing is correct.

Reply:
Is the slice ever going to container more than one entry in it?

Reply:
Yep - `MetricsConsolidator` spits out a slice on Drain/Flush